### PR TITLE
Add Upload saver

### PR DESCRIPTION
**Pull request by jacobkiers**
I have a use case where I can't run a MongoDB installation directly with my application, and uploading and importing profiles is relatively hard.

Therefore, I have added an extra saver, which makes it possible to upload a profile to the XHGui application.

The PR for XHGui itself is also on its way: https://github.com/perftools/xhgui/pull/230

### CODE REVIEW

File: config/config.default.php [hunk 1]
```diff
@@ -11,16 +11,20 @@
...
+
+    // For file
+    //
+    //'save.handler' => 'file',
+    //'save.handler.filename' => dirname(__DIR__) . '/cache/' . 'xhgui.data.' . microtime(true) . '_' . substr(md5($url), 0, 6),
 
-    // Needed for file save handler. Beware of file locking. You can adujst this file path
-    // to reduce locking problems (eg uniqid, time ...)
-    //'save.handler.filename' => __DIR__.'/../data/xhgui_'.date('Ymd').'.dat',
+    // For url
```
Comment (markstory) [inline config/config.default.php:21]: Should this say 'upload' instead of url?

File: src/Xhgui/Saver/Upload.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        $headers = array(
+            'Accept: application/json',         // Prefer to receive JSON back
+            'Content-Type: application/json'    // The sent data is JSON
+        );
+
+        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'POST');
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($ch, CURLOPT_POSTFIELDS, $json);
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);
+        curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
```
Comment (markstory) [inline src/Xhgui/Saver/Upload.php:28]: What about a timeout? Do you want to lock up the host application while this request sits around?

**Comment by jacobkiers** [comment]
Hi, just quick note to assure you that I haven't forgotten about this. I'm currently unable to work on it, but I expect to be able to address the comments within a month.

File: src/Xhgui/Saver/Upload.php (hunk 2 above)
Comment (glensc) [reply src/Xhgui/Saver/Upload.php:28]: timeout was added in https://github.com/perftools/xhgui-collector/commit/790cfdb08763de78d5cd12e98223832e286c39fe

**Pull request merged**